factory-containers/assemble-system-image

```diff
@@ -69,6 +169,14 @@ def copy_container_images_to_wic(target: dict, app_image_dir: str, wic_image: st
     containers_arch = arch_map[target['custom']['arch']]
 
     app_image_tar_src = os.path.join(app_image_dir, containers_sha, '{}-{}.tar'.format(containers_sha, containers_arch))
+    if not os.path.exists(app_image_tar_src):
+        logger.info('Container images have not been found, trying to obtain them...')
+        os.makedirs(os.path.dirname(app_image_tar_src), exist_ok=True)
+        dump_apps_container_images(target, containers_arch, app_image_tar_src, token)
+
+    if not os.path.exists(app_image_tar_src):
```

Comment:
can this condition happen? the if above for that same condition calls `makedirs`?

Reply:
Yes, the above `makedirs` makes a dir while the given `if` verifies if the tar file exists, I am just not sure if in all failure cases exception will be generated so double-checking if the tar file containing container images actually exists.

Reply:
I think it should. If it doesn't the code will fail but should hopefully be debuggable. I'd do one of two things 1) remove this 2) put a comment that explains why.

Reply:
The problem is that the following code won't fail if `app_image_tar_src` doesn't exist and it's really bad because it will look like the run/job is successful while the resultant wic image won't contain container images.
For some reason this call `subprocess.check_call([wic_tool, 'cp', app_image_tar_src, app_image_tar_dst])` is successful if `app_image_tar_src` does not exists. I'll double check it tomorrow.

Reply:
Ugh - i read this wrong. the first you do a `mkdir(dirname(app_image_tar_src)` you then check if `app_image_tar_src` exists. that's two different things. sorry for the disctraction.